src/coreclr/src/vm/arm64/asmhelpers.asm

```diff
@@ -1478,6 +1478,39 @@ __HelperNakedFuncName SETS "$helper":CC:"Naked"
 
 #endif
 
+;;-----------------------------------------------------------------------------
+;; The following helper will access ("probe") a word on each page of the stack
+;; starting with the page right beneath sp down to the one pointed to by x9.
+;; The procedure is needed to make sure that the "guard" page is pushed down below the allocated stack frame.
+;; The call to the helper will be emitted by JIT in the function/funclet prolog when large stack frame is required.
+;;-----------------------------------------------------------------------------
+; On entry:
+;   x9   - points to the lowest address on the stack frame being allocated (i.e. [InitialSp - FrameSize])
+;   sp   - points to some byte on the last probed page
+; On exit:
+;   x9   - is preserved
```

Comment:
Should you mention x30 is trashed?

Reply:
`x30` is never trashed. `x30` is the same register as `lr`. I am using `x30` name instead of `lr` since the register is not used as *link register* and I wanted to emphasize that.